### PR TITLE
Make outbreak detection possible without labels

### PR DESCRIPTION
**Pull request by JarnoRFB**

### CODE REVIEW

File: docs/3_user_guide.rst [hunk 1]
```diff
@@ -43,13 +43,9 @@ subtracted from the total cases to obtain the *in control*
...
-counts you can fill the ``n_outbreak_cases`` column with
-zeros under the assumption that your data is in fact
-*in control* data.
-
-.. code-block:: python
-
-    data["n_outbreak_cases"] = 0
+counts ``epysurv`` implicitly adds the ``n_outbreak_cases`` column filled
+with zeros under the assumption that your data is in fact
+*in control* data. A waring is still issued in this case.
```
Comment (aauss) [inline docs/3_user_guide.rst]: A warning

File: epysurv/models/timepoint/_base.py [hunk 2]
```diff
@@ -36,9 +37,16 @@ class TimepointSurveillanceAlgorithm:
 
     def fit(self, data: pd.DataFrame) -> "TimepointSurveillanceAlgorithm":
         """Expects data with time series index, case counts and outbreak labels."""
-        # Remove outbreaks cases from baseline.
+        self._validate_data(data)
         data = data.copy()
-        data.n_cases -= data.n_outbreak_cases
+        if "n_outbreak_cases" in data.columns:
+            # Remove outbreaks cases from baseline.
+            data.n_cases -= data["n_outbreak_cases"]
```
Comment (aauss) [inline epysurv/models/timepoint/_base.py]: What if `n_cases` and `n_outbreak_cases` are already separated by the user? Then subtracting would falsify `n_cases`.

File: epysurv/models/timepoint/_base.py [hunk 3]
```diff
@@ -110,7 +118,7 @@ def predict(self, data: pd.DataFrame) -> pd.DataFrame:
         super().predict(data)
         # Concat training and prediction data. make index array for range param.
```
Comment (aauss) [inline epysurv/models/timepoint/_base.py]: data. Make index..

File: epysurv/models/timepoint/glr.py [hunk 4]
```diff
@@ -46,7 +46,7 @@ class GLRNegativeBinomial(STSBasedAlgorithm):
         - ("dec",) only decreases are regarded
         - ("inc", "dec") both increases and decreases are regarded.
     upperbound_statistic
-        a string specifying the type of upperbound-statistic that is returned.
+        A string specifying the type of upperbound-statistic that is returned.
         With "cases" the number of cases that would have been necessary
```
Comment (aauss) [inline epysurv/models/timepoint/glr.py]: "cases" for the number of cases that would have been necessary to produce an alarm."value" returns the GLR-statistic.

File: tests/test_timepoint_models.py [hunk 5]
```diff
@@ -93,3 +93,20 @@ def test_output_format(train_data, test_data):
...
+
+def test_validate_data_on_fit(train_data):
+    model = EarsC1()
+    with pytest.raises(ValueError):
+        model.fit(train_data.rename(columns={"n_cases": "wrong_column_name"}))
+
+
+@pytest.mark.parametrize("Algo", algos_to_test)
+def test_prediction_witout_labels(train_data, test_data, shared_datadir, Algo):
+    """Test only works, because sample data contains no outbreaks in the training set."""
```
Comment (aauss) [inline tests/test_timepoint_models.py]: Test should only pass if...

File: epysurv/models/timepoint/_base.py (hunk 2 above)
Comment (JarnoRFB) [reply epysurv/models/timepoint/_base.py]: The semantics of `n_cases` is the total number of cases (outbreak + non outbreak). So I do not see a problem here.

File: tests/test_timepoint_models.py (hunk 5 above)
Comment (JarnoRFB) [reply tests/test_timepoint_models.py]: I meant to justify the fact that the test runs with the same data as `test_prediction`. I tried to make it a bit more explicit now.

File: epysurv/models/timepoint/_base.py (hunk 2 above)
Comment (aauss) [reply epysurv/models/timepoint/_base.py]: Ah, I found it in the user guide.

File: docs/3_user_guide.rst [hunk 6]
```diff
@@ -43,13 +43,9 @@ subtracted from the total cases to obtain the *in control*
...
 If you do not have any labeled outbreak data, but just the raw
-counts you can fill the ``n_outbreak_cases`` column with
-zeros under the assumption that your data is in fact
-*in control* data.
-
-.. code-block:: python
-
-    data["n_outbreak_cases"] = 0
+counts ``epysurv`` implicitly adds the ``n_outbreak_cases`` column filled
+with zeros under the assumption that your data is in fact
```
Comment (aauss) [inline docs/3_user_guide.rst]: I thought that NaNs would be better since zeros suggest that we "know" that there are not outbreak cases but maybe we just don't know

Comment (JarnoRFB) [reply docs/3_user_guide.rst]: Doing it with NaNs is not possible, as you could not subtract them from the `n_cases` column. What happens is that simply nothing is subtracted from the `n_cases` column. But maybe the wording is a still a bit confusing.
